plugins/woocommerce/client/blocks/assets/js/base/context/hooks/payment-methods/use-payment-method-interface.ts

```diff
@@ -56,61 +56,74 @@ export const usePaymentMethodInterface = (): PaymentMethodInterface => {
 				customerId: store.getCustomerId(),
 				isCalculating: store.isCalculating(),
 			};
-		} );
-	const { paymentStatus, activePaymentMethod, shouldSavePayment } = useSelect(
-		( select ) => {
-			const store = select( paymentStore );
+		}, [] );
 
-			return {
-				// The paymentStatus is exposed to third parties via the payment method interface so the API must not be changed
-				paymentStatus: {
```

Comment:
This object caused the problem because it changes with every re-render. Fix was simply to move each status to top level and construct the object later on.